### PR TITLE
update

### PR DESCRIPTION
**Pull request by weihuan1111**

### CODE REVIEW

**Comment by weihuan1111** [comment]
@greens please help to review the merge require,thanks

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 1]
```diff
@@ -269,9 +270,12 @@ def parse(String description) {
...
 					descMap.clusterInt == zigbee.ONOFF_CLUSTER ||
 					descMap.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER ||
-					descMap.clusterInt == CLUSTER_WINDOW_COVERING) {
+					descMap.clusterInt == CLUSTER_WINDOW_COVERING && (!isThirdReality())) {
 				event = getButtonEvent(descMap)
-			}
+			} else if ((descMap.clusterInt == zigbee.ONOFF_CLUSTER ||
+			    descMap.clusterInt == 0x0012 )&&(isThirdReality())) {
+				    event = getButtonEvent(descMap)
+				}
```
Comment (greens) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: this separate if statement is not required after the refactoring to move all the code into `getButtonEvent`

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 2]
```diff
@@ -336,6 +340,13 @@ private Map getButtonEvent(Map descMap) {
 						: [state: "held", buttonNumber: REMOTE_BUTTONS.LEFT] },
 					 0x09: { [state: "", buttonNumber: 0] }]
 	]
+	
+	Map ThirdRealityButtonMapping = [
```
Comment (greens) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: this looks unused

**Review by greens — CHANGES_REQUESTED** [review]
In addition to my other comments, you have mixed tabs and spaces in several places

**Comment by weihuan1111** [comment]
@greens please help to review the merge require,thanks

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 3]
```diff
@@ -395,7 +397,18 @@ private Map getButtonEvent(Map descMap) {
...
+	    if (descMap.clusterInt == 0x0012) {
+		    buttonNumber = 1
+			if (descMap.value == "0002") {
+				buttonState = "double"
+			} else if (descMap.value == "0001") {
+				buttonState = "pushed"
+			} else if (descMap.value == "0000") {
+				buttonState = "held"
+			}
+		}
```
Comment (greens) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: you still have mixed whitespace

**Comment by weihuan1111** [comment]
@greens I have update the code.

**Review by greens — APPROVED** [review]

**Pull request merged**